lib/registry.js

```diff
@@ -127,10 +129,34 @@ exports.load = function(registry) {
               if (!active)
                 return;
               active = false;
-              checkRetry(err);
+              return checkConfigure(err) || checkRetry(err);
             });
           }
 
+          function checkConfigure(err) {
+            if (err && err.config) {
```

Comment:
Again perhaps a comment to describe what this function does? Something like "When err.config is set, that indicates config credentials are somehow the cause. Call the configure hook and reinstantiate the registry with new config"